test/LaunchDarkly.EventSource.Tests/Stubs/StubHandler.cs

```diff
@@ -57,140 +58,141 @@ public HttpRequestMessage AwaitRequest()
-        {
-            this.message = message;
-        }
-
-        override public HttpResponseMessage MakeResponse(CancellationToken cancellationToken)
-        {
-            return message;
-        }
+    internal class StubResponseWithHttpResponse : StubResponse
+    {
+        private readonly HttpResponseMessage message;
+
+        public StubResponseWithHttpResponse(HttpResponseMessage message)
+        {
+            this.message = message;
+        }
+
+        override public HttpResponseMessage MakeResponse(CancellationToken cancellationToken)
+        {
+            return message;
+        }
     }
 
-    internal class StubResponseWithIOError : StubResponse
-    {
-        override public HttpResponseMessage MakeResponse(CancellationToken cancellationToken)
-        {
-            throw new HttpRequestException("Unit Test Exception Message");
-        }
+    internal class StubResponseWithIOError : StubResponse
+    {
+        override public HttpResponseMessage MakeResponse(CancellationToken cancellationToken)
+        {
+            throw new HttpRequestException("Unit Test Exception Message");
+        }
     }
 
-    internal class StubResponseWithStream : StubResponse
-    {
-        private readonly StreamAction[] _actions;
-        private readonly Encoding _encoding;
-        private readonly bool _specifiedEncoding;
-
-        public StubResponseWithStream(StreamAction[] actions, Encoding encoding = null)
-        {
-            _actions = actions;
-            _encoding = encoding ?? Encoding.UTF8;
-            _specifiedEncoding = encoding != null;
-        }
-
-        override public HttpResponseMessage MakeResponse(CancellationToken cancellationToken)
-        {
-            var httpResponse = new HttpResponseMessage(HttpStatusCode.OK);
-            var streamRead = new AnonymousPipeServerStream(PipeDirection.In);
-            var streamWrite = new AnonymousPipeClientStream(PipeDirection.Out, streamRead.ClientSafePipeHandle);
-            var content = new StreamContent(streamRead);
-            content.Headers.ContentType = new MediaTypeHeaderValue("text/event-stream");
+    internal class StubResponseWithStream : StubResponse
+    {
+        private readonly StreamAction[] _actions;
+        private readonly Encoding _encoding;
+        private readonly bool _specifiedEncoding;
+
+        public StubResponseWithStream(StreamAction[] actions, Encoding encoding = null)
+        {
+            _actions = actions;
+            _encoding = encoding ?? Encoding.UTF8;
+            _specifiedEncoding = encoding != null;
+        }
+
+        override public HttpResponseMessage MakeResponse(CancellationToken cancellationToken)
+        {
+            var httpResponse = new HttpResponseMessage(HttpStatusCode.OK);
+            var pipe = new Pipe();
+            var readStream = pipe.Reader.AsStream(true);
+            var content = new StreamContent(readStream);
```

Comment:
This part is the most significant change. For the tests that don't need a real embedded HTTP server, we use this fake HTTP handler that pipes data into the stream. I was using this other pipe mechanism (see old lines 131-132) which is really intended for IPC, and it seems that it doesn't behave very predictably when you use it within a single process: the behavior I saw was that in Windows, reading from `streamRead` when there wasn't yet any more data waiting would not block as it should, but would return "0 bytes" as if the stream was actually finished. And that would cause EventSource to think the server had closed the request, which would mess up tests in various ways.

Anyway, using `Pipe` from `System.IO.Pipelines` makes this behave as it should.